Wrappers/Python/ccpi/viewer/ui/dialogs.py

```diff
@@ -137,6 +185,125 @@ def enableDisableDimZ(self):
             widgets['dim_Z_field'].setEnabled(True)
         else:
             widgets['dim_Z_field'].setEnabled(False)
+    
+    def preview(self):
+        pars = self.getRawAttrs()
+        
+        # retrieve info about image file from interface
+        dimensionality = [3, 2][self.getWidget('dimensionality').currentIndex()]
+        dimX, dimY, dimZ = pars['shape']
+        isFortran = pars['is_fortran']
+        isBigEndian = pars['is_big_endian']
+        # typecode = pars['typecode']
+        typecode = self.getWidget('dtype').currentText()
+
+        
+        if isFortran:
+            shape = (dimX, dimY)
+        else:
+            shape = (dimY, dimX)
+        if dimensionality == 3:
+            if isFortran:
+                shape = (dimX, dimY, dimZ)
+            else:
+                shape = (dimZ, dimY, dimX)
+
+
+        
+        
+        # Construct a data type
+        dt = np.dtype(typecode)
+        
+        if isBigEndian:
+            dt_txt = ">" # big endian
+        else:
+            dt_txt = "<"
+        dt = dt.newbyteorder(dt_txt)
+
+        bytes_per_element = dt.itemsize
+
+        # basic sanity check
+        file_size = os.stat(self.fname).st_size
+
+        expected_size = reduce (lambda x,y: x*y, shape, 1) * bytes_per_element        
+        
+        if file_size != expected_size:
+            errors = {"type": "size", 
+                      "file_size": file_size,
+                      "expected_size": expected_size}
+            return (errors)
+
+        # read centre slice
+        offset = 0
+        slice_size = -1
+        if dimensionality == 3:
+            # read the centre slice
+            slice_size = shape[1]*shape[0]
+            offset = shape[2]*slice_size//2
+
+        rawfname = os.path.join(tempfile.gettempdir(),"test.raw")
+        
+        offset = offset * bytes_per_element
+        slices_to_read = 1
+        if shape[2] > 1:
+            slices_to_read = 2
+        with open(self.fname, 'br') as f:
+            f.seek(offset)
+            raw_data = f.read(slice_size*bytes_per_element* slices_to_read)
+            with open(rawfname, 'wb') as f2:
+                f2.write(raw_data)
+
+        reader2 = vtk.vtkImageReader2()
+        reader2.SetFileName(rawfname)
+
+        vtktype = Converter.dtype_name_to_vtkType[dt.name]
+        reader2.SetDataScalarType(vtktype)
+
+        if isBigEndian:
+            reader2.SetDataByteOrderToBigEndian()
+        else:
+            reader2.SetDataByteOrderToLittleEndian()
+
+        reader2.SetFileDimensionality(len(shape))
+        vtkshape = shape[:]
+        if not isFortran:
+            # need to reverse the shape (again)
+            vtkshape = shape[::-1]
```

Comment:
I used the `cilRawCroppedReader` but it gave the wrong output. I updated it now with the code that I had implemented above. This will require some testing, which I haven't done.

Reply:
@paskino is this conversation resolved or should we do more tests?